### PR TITLE
Run GLS on real phenotypes from PhenomeXcan and eMERGE

### PR DESCRIPTION
**Pull request by miltondp**
This PR adds the results of running the GLS model in PhenoPLIER on two TWAS resources on real data (PhenomeXcan and eMERGE). Many file changes are binary files with the results (only in PhenomeXcan) and others are scripts for cluster jobs and notebooks with some analyses.

### CODE REVIEW

File: nbs/15_gsa_gls/30-real_phenotypes/05-group_gwas_by_n_variants.ipynb [hunk 1]
```diff
@@ -0,0 +1,1599 @@
+{
```
Comment (nrosed) [inline nbs/15_gsa_gls/30-real_phenotypes/05-group_gwas_by_n_variants.ipynb:1]: <p>I dont 100% follow whats going on here. I think it's fine if this is just a helper notebook, but I think just a little bit more description here can help a lot.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/phenoplier/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (nrosed) [inline nbs/15_gsa_gls/30-real_phenotypes/05-group_gwas_by_n_variants.ipynb:1]: <p>what is rapid and astle?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/phenoplier/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (nrosed) [inline nbs/15_gsa_gls/30-real_phenotypes/05-group_gwas_by_n_variants.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    assert df_stats["min"] &gt; 0.99
</pre><p>why do df_stats need to be larger than 0.99?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/phenoplier/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='60'/>

Comment (nrosed) [inline nbs/15_gsa_gls/30-real_phenotypes/05-group_gwas_by_n_variants.ipynb:1]: <p>why is this section commented out?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/phenoplier/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='63'/>

File: nbs/15_gsa_gls/30-real_phenotypes/10-summarize-phenomexcan.ipynb [hunk 2]
```diff
@@ -0,0 +1,2001 @@
+{
```
Comment (nrosed) [inline nbs/15_gsa_gls/30-real_phenotypes/10-summarize-phenomexcan.ipynb:1]: <p>what is GLS?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/phenoplier/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (nrosed) [inline nbs/15_gsa_gls/30-real_phenotypes/10-summarize-phenomexcan.ipynb:1]: <p>whoa~! lots of significant hits, is this expected?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/phenoplier/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

File: nbs/15_gsa_gls/30-real_phenotypes/README.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+# load in bash session all PhenoPLIER environmental variables
+eval `python ~/projects/phenoplier/libs/conf.py`
+
+# make sure they were loaded correctly
+# should output something like /project/...
+echo $PHENOPLIER_ROOT_DIR
+```
+
+### Desktop computer
```
Comment (nrosed) [inline nbs/15_gsa_gls/30-real_phenotypes/README.md:34]: is it possible to run this on a desktop computer? just curious

**Comment by nrosed** [comment]
looks good! I don't know a lot of the details of the project, but the code looks good. Added a few comments regarding the descriptions for notebooks. Also the qq-plots look like they have a lot of significant hits -- is that expected? I also like to add confidence intervals to my qq-plots, but just my personal preference. I know how to add it in R, but not python.

```   
    ### plot_df is a data.frame with a column of p-values, each row is a test result
    n = nrow(plot_df)
    plot_df$idx = -1*log10((1:n)/n)
    plot_df$clower = -log10(qbeta(p = (1 + 0.95) / 2, shape1 = 1:n, shape2 = n:1))
    plot_df$cupper = -log10(qbeta(p = (1 - 0.95) / 2, shape1 = 1:n, shape2 = n:1))


    gg_qq_null = ggplot(plot_df, aes(x=idx, y=ROCR)) +  # only use black
        geom_abline(intercept = 0, slope = 1) +
        geom_point() +
        geom_line(aes(idx, cupper), linetype = 2) +
        geom_line(aes(idx, clower), linetype = 2)+
        ggtitle(outname) +
        theme_bw() + labs(x ="Expected (-log10)", y = "Observed (-log10)")
```

**Review by nrosed — APPROVED** [review]

File: nbs/15_gsa_gls/30-real_phenotypes/05-group_gwas_by_n_variants.ipynb (hunk 1 above)
Comment (miltondp) [reply nbs/15_gsa_gls/30-real_phenotypes/05-group_gwas_by_n_variants.ipynb:1]: Done!

Comment (miltondp) [reply nbs/15_gsa_gls/30-real_phenotypes/05-group_gwas_by_n_variants.ipynb:1]: They are cohorts (sets of GWAS results with the same set of variants). I updated the description.

Comment (miltondp) [reply nbs/15_gsa_gls/30-real_phenotypes/05-group_gwas_by_n_variants.ipynb:1]: These `assert`s are my visual assertions. In this case, I want to make sure that if I run the notebook again, those pairs of traits share at least 0.99 of the variants.

Comment (miltondp) [reply nbs/15_gsa_gls/30-real_phenotypes/05-group_gwas_by_n_variants.ipynb:1]: For future reference. I added a description.

File: nbs/15_gsa_gls/30-real_phenotypes/10-summarize-phenomexcan.ipynb (hunk 2 above)
Comment (miltondp) [reply nbs/15_gsa_gls/30-real_phenotypes/10-summarize-phenomexcan.ipynb:1]: Generalized Least Squares. I updated the description.

Comment (miltondp) [reply nbs/15_gsa_gls/30-real_phenotypes/10-summarize-phenomexcan.ipynb:1]: <p>I don't know if it is expected, but it's similar to other large runs on multiple phenotypes (4,091 here). There is another PR with the null simulations also, where we found that the model is approximately well calibrated and with mean type I errors around 0.05.</p><p> </p><p>Some of the LV-trait associations make a lot of sense, like LV246, with genes expressed in adipose tissue, and associated to many cardiovascular traits.</p>

**Comment by miltondp** [comment]
<p>Wow, thank you so much for this, Natalie!</p>

File: nbs/15_gsa_gls/30-real_phenotypes/README.md (hunk 3 above)
Comment (miltondp) [reply nbs/15_gsa_gls/30-real_phenotypes/README.md:34]: Yes, it is! Sorry for the delay in replying, I thought I responded to all your comments!

File: nbs/15_gsa_gls/30-real_phenotypes/README.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+  
+  mkdir -p _tmp/phenoplier/phenomexcan
+  mkdir -p _tmp/phenoplier/emerge
+  
+  cat $cluster_job_file | envsubst '${pheno_id}' | ${PHENOPLIER_JOBS_EXECUTOR}
+}
+
+export -f run_job
+
+# (optional) export function definition so it's included in the Docker container
```
[the remark this answers is not in the document]
Comment (miltondp) [reply nbs/15_gsa_gls/30-real_phenotypes/README.md:90]: Yes, through Singularity. But I still didn't test it thoroughly.

**Pull request merged**